lib/api/npm.js

```diff
@@ -7,25 +7,31 @@ const registryAuthToken = require('registry-auth-token');
 const logger = require('winston');
 
 module.exports = {
+  setNpmrc,
   getDependency,
   resetCache,
 };
 
 let npmCache = {};
+let npmrc = null;
 
 function resetCache() {
   npmCache = {};
 }
 
+async function setNpmrc(input) {
+  npmrc = input;
+}
+
 async function getDependency(name) {
   logger.debug(`getDependency(${name})`);
   const scope = name.split('/')[0];
-  const regUrl = registryUrl(scope);
+  const regUrl = registryUrl(scope, { npmrc });
```

Comment:
These configuration options are always passed to `registryUrl` and `registryAuthToken`, but if they are null then the file system will be used instead.

Reply:
@rarkins This line does not work w/ a private npm registry when the `.npmrc` is read from target repository.  Example:

```
npmrc = { registry: 'https://my.corporate.local/npm-private/' }
regUrl = https://registry.npmjs.org/
pkgUrl = https://registry.npmjs.org/sort-object
```

It looks like [registry-url does not read that second paramter](https://github.com/sindresorhus/registry-url/blob/master/index.js)

The private npm registry does work, when I put the .npmrc file to the renovate deployment folder, i.e.:

```
|- node_modules
   |- .bin/renovate
|- .npmrc
```

Reply:
When read from that second `.npmrc`, the url is constructed right:

```
regUrl https://my.corporate.local/npm-private/
pkgUrl https://my.corporate.local/npm-private/sort-object
{ registry: 'https://my.corporate.local/npm-private/' }
```

It works when there are no private packages references. When renovate tries to read npm packages that are only available on the private registry, it errors.

Reply:
@dherges thanks. could you raise this as a new issue, so we can discuss there? I'll ask questions to clarify then, as otherwise they might get a little lost inside this older PR

Reply:
FYI the code for this has changed just recently, although it may be "differently wrong". I see what you mean about how the registry appears to be being ignored

Reply:
I opened #763